### PR TITLE
entities, towers, enemies, gameplay scene, wave manager, entity manager, wave.json

### PR DESCRIPTION
**Pull request by froghood**

### CODE REVIEW

File: TowerOffense.Objects/Base/Tower.cs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            int titleBarHeight = 24,
+            int borderThickness = 1) : base(
+                scene,
+                entityManager,
+                size,
+                position,
+                titleBarHeight,
+                borderThickness) {
+
+            //TODO: implement
```
Comment (alex-mike-mark-again) [inline TowerOffense.Objects/Base/Tower.cs:42]: iirc thiis doesn't need to be todone.

File: TowerOffense.Scenes/Example/Objects/Worm.cs [hunk 2]
```diff
@@ -32,17 +32,17 @@ public class Worm : SceneWindow {
...
+            Position += new Vector2() {
                 X = MathF.Cos(_angle) * _speed,
                 Y = MathF.Sin(_angle) * _speed
             };
 
-            _position = new Vector2() {
-                X = MathF.Max(0f, MathF.Min(_position.X, 1920f - Size.X)),
-                Y = MathF.Max(0f, MathF.Min(_position.Y, 1080f - Size.Y))
-            };
+            // _position = new Vector2() {
```
Comment (alex-mike-mark-again) [inline TowerOffense.Scenes/Example/Objects/Worm.cs:40]: more commented out stuff

File: TowerOffense.Objects/Base/SceneWindow.cs [hunk 3]
```diff
@@ -10,11 +10,27 @@ namespace TowerOffense.Objects.Base {
...
 
 
-        public Point Position {
-            get => new Point(_form.Location.X, _form.Location.Y);
-            set => _form.Location = new System.Drawing.Point(value.X, value.Y);
+        public Vector2 Position { get; set; }
+
+        public Vector2 Offset { get; set; }
+
+        // public Point Position {
```
Comment (alex-mike-mark-again) [inline TowerOffense.Objects/Base/SceneWindow.cs:17]: more commented stuff :c

File: TowerOffense.Objects/Base/SceneWindow.cs [hunk 4]
```diff
@@ -126,16 +158,13 @@ public abstract class SceneWindow : SceneObject {
...
         }
 
-
-
         public override void Update(GameTime gameTime) {
 
             // window dragging & close button
             if (!Draggable) _isBeingDragged = false;
-            if (_isBeingDragged) _form.Location = new System.Drawing.Point(Cursor.Position.X - _dragOffset.X, Cursor.Position.Y - _dragOffset.Y);
+            //if (_isBeingDragged) _form.Location = new System.Drawing.Point(Cursor.Position.X - _dragOffset.X, Cursor.Position.Y - _dragOffset.Y);
```
Comment (alex-mike-mark-again) [inline TowerOffense.Objects/Base/SceneWindow.cs:167]: commented...

**Review by alex-mike-mark-again — APPROVED** [review]
Looks good to me other than a bunch of commented stuff.

File: TowerOffense.Objects/Enemies/SpiderEnemy.cs [hunk 5]
```diff
@@ -71,7 +72,7 @@ bool fromPortal
                     }
                     break;
                 case EnemyState.Attacking:
-                    if (StateTime >= 2f) {
+                    if (StateTime >= 1.25f) {
```
Comment (alex-mike-mark-again) [inline TowerOffense.Objects/Enemies/SpiderEnemy.cs:75]: This smells like the kind of thing that should either be a constant or a field in the class.

Comment (froghood) [reply TowerOffense.Objects/Enemies/SpiderEnemy.cs:75]: lazy sorry

**Pull request merged**